### DISCUSSION TITLE
List files API?

### DISCUSSION THREAD

**Discussion by dhrumil83** [microsoft/Microsoft-Fabric-workload-development-sample#61] [Q&A] [answered]
This doc mentions a list tables API: https://learn.microsoft.com/en-us/rest/api/fabric/lakehouse/tables/list-tables?tabs=HTTP

Wondering if there is a similar API to list all files? Looking for an example that displays (for example) all csv files in the selected Lakehouse.

**Comment by IdanGoldfarb1989** [comment, marked as answer]
Lakehouse public APIs do not include this feature. However, you can use Azure Storage APIs, which also work on OneLake items, to retrieve this information.

For more details, see:
[Azure Storage APIs - List Paths](https://learn.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/list?view=rest-storageservices-datalakestoragegen2-2019-12-12)

You may need to filter the results to retrieve only CSV files. We also have an example of using this API in our sample code:
[Sample Code](https://github.com/microsoft/Microsoft-Fabric-developer-sample/blob/1f0ad5eb514ac756ba7a9d5ecd5abdf859081e68/Backend/src/Services/LakeHouseClientService.cs#L293C50-L293C61)

**Reply by dhrumil83** [reply]
Thanks. I cannot access the link above: https://github.com/microsoft/Microsoft-Fabric-developer-sample/blob/1f0ad5eb514ac756ba7a9d5ecd5abdf859081e68/Backend/src/Services/LakeHouseClientService.cs#L293C50-L293C61

**Reply by IdanGoldfarb1989** [reply]
Search for 'GetPathList' function inside LakeHouseClientService.cs
